test/functional/e2e/user-management/user-profile.e2e-spec.ts

```diff
@@ -0,0 +1,166 @@
+  });
+
+  afterAll(async () => {
+    const requestUserData = await getUser(email);
+    userId = requestUserData.body.data.user.id;
+    await removeUserMutation(userId);
+    await browser.close();
+  });
+
+  describe('User profile', () => {
+    test('User updates its profile successfully', async () => {
+      await userProfilePage.verifyUserProfileForm(page);
+      await userProfilePage.updateUserProfileFields(
+        page,
+        fullNameChange,
+        firstNameChange,
+        lastNameChange,
+        countryName,
+        city,
+        phone,
+        bio
+      );
+      await userProfilePage.updateSkillsTagsEditProfilePage(page, skills);
+      await userProfilePage.updateKeywordsTagsEditProfilePage(page, keywords);
+      await userProfilePage.addReferenceEditProfilePage(
+        page,
+        referenceName,
+        referenceValue
+      );
+      await userProfilePage.saveChangesPofilePage(page);
+      await userProfilePage.closeSuccessMessageProfilePage(page);
+      await userProfilePage.closeEditProfilePage(page);
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        email
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        bio
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        phone
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        countryName
+      );
+      expect(await userProfilePage.verifyTagsEntities(page)).toContain(skills);
+      expect(await userProfilePage.verifyTagsEntities(page)).toContain(
+        keywords
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        referenceName
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        referenceValue
+      );
+    });
+
+    test('User removes its reference successfully', async () => {
+      await userProfilePage.removeReferenceEditProfilePage(page);
+      await userProfilePage.saveChangesPofilePage(page);
+      await userProfilePage.closeSuccessMessageProfilePage(page);
+      await userProfilePage.closeEditProfilePage(page);
+
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        email
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        bio
+      );
+      expect(await userProfilePage.verifyUserProfileEntities(page)).toContain(
+        phone
+      );
+      expect(await userProfilePage.verifyTagsEntities(page)).toContain(skills);
+      expect(await userProfilePage.verifyTagsEntities(page)).toContain(
+        keywords
+      );
+      expect(
+        await userProfilePage.verifyUserProfileEntities(page)
+      ).not.toContain(referenceName);
+      expect(
+        await userProfilePage.verifyUserProfileEntities(page)
+      ).not.toContain(referenceValue);
```

Comment:
You can save the result from `verifyUserProfileEntities` into a variable and then execute each check against that variable instead of calling `verifyUserProfileEntities`. `verifyUserProfileEntities` will return same results every time, but will have to evaluate the page, select element of that page enumerate the elements etc. which is unnecessary. The only reason to call  `verifyUserProfileEntities` more than once during the assertion process is if an action was executed between two assertions and that action should change the output of `verifyUserProfileEntities`.